src/main/java/Duke.java

```diff
@@ -1,10 +1,117 @@
+        System.out.println(message.toString());
+        System.out.println("Now you have " + (index + 1) + " tasks in the list");
+        printHorizontalLine();
+    }
+    
+    //echo the list
+    public static void echoList(Task[] tasks) {
+        printHorizontalLine();
+        int index = 0;
+        while (tasks[index] != null){
+            System.out.println((index + 1) + ". " + tasks[index].toString());
+            index++;
+        }
+
+        System.out.println();
+        printHorizontalLine();
+    }
+
+    //change the status of the task
+    public static void changeStatus(Task task) {
+        printHorizontalLine();
+        System.out.println("Nice! I've marked this task as done:");
+        
+        task.setIsDone(true);
+        System.out.println(" [" + task.getStatusIcon() + "] " + task.description);
+
+        System.out.println();
+        printHorizontalLine();
+    }
+
+    public static Task createType(String message) {
+        String[] infoEntered = message.split(" ");
+        int nameLength;
+        switch (infoEntered[0]) {
+        case ("todo"):
+            nameLength = 4;
+            return new ToDo(message.substring(nameLength + 1));
+        case ("deadline"):
+            nameLength = 8;
+            int indexBy = message.indexOf("/");
+            return new Deadline(message.substring(nameLength + 1, indexBy - 1),
+                    message.substring(indexBy + 1));
+        case ("event"):
+            nameLength = 5;
+            int indexAt = message.indexOf("/");
+            return new Event(message.substring(nameLength + 1, indexAt - 1),
+                    message.substring(indexAt + 1));
+        default:
+            return null;
+        }
+    }
     public static void main(String[] args) {
+        
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        
+        Scanner in = new Scanner(System.in);       
+        
+        printWelcomeMessage();
+        
+        Task[] tasks = new Task[MAX_SIZE];
+        int index = 0;
+        String enteredMessage = in.nextLine();
+
+        while(!enteredMessage.equals("bye")){
+            if (enteredMessage.equals("list")){
+                echoList(tasks);
+            } else if (enteredMessage.startsWith("done")) {
+                int numberIndex = 5;
+                int itemIndex = Integer.parseInt(enteredMessage.substring(numberIndex)) - 1;
+                changeStatus(tasks[itemIndex]);
+            } else {
+                tasks[index] = createType(enteredMessage);
+                echoMessage(tasks[index], index);
+                index++;
+            }
+            enteredMessage = in.nextLine();
+        }
+
+        printByeMessage();
     }
```

Comment:
I like how your code is not deeply nested so there is no "arrow head" code.

Reply:
I like this too!